stable_baselines/td3/td3.py

```diff
@@ -0,0 +1,475 @@
+                    if len(ep_info_buf) > 0 and len(ep_info_buf[0]) > 0:
+                        logger.logkv('ep_rewmean', safe_mean([ep_info['r'] for ep_info in ep_info_buf]))
+                        logger.logkv('eplenmean', safe_mean([ep_info['l'] for ep_info in ep_info_buf]))
+                    logger.logkv("n_updates", n_updates)
+                    logger.logkv("current_lr", current_lr)
+                    logger.logkv("fps", fps)
+                    logger.logkv('time_elapsed', int(time.time() - start_time))
+                    if len(episode_successes) > 0:
+                        logger.logkv("success rate", np.mean(episode_successes[-100:]))
+                    if len(infos_values) > 0:
+                        for (name, val) in zip(self.infos_names, infos_values):
+                            logger.logkv(name, val)
+                    logger.logkv("total timesteps", self.num_timesteps)
+                    logger.dumpkvs()
+                    # Reset infos:
+                    infos_values = []
+            return self
+
+    def action_probability(self, observation, state=None, mask=None, actions=None, logp=False):
+        _ = np.array(observation)
+
+        if actions is not None:
+            raise ValueError("Error: TD3 does not have action probabilities.")
+
+        # here there are no action probabilities, as DDPG does not use a probability distribution
+        warnings.warn("Warning: action probability is meaningless for TD3. Returning None")
+        return None
+
+    def predict(self, observation, state=None, mask=None, deterministic=True):
+        observation = np.array(observation)
+        vectorized_env = self._is_vectorized_observation(observation, self.observation_space)
+
+        observation = observation.reshape((-1,) + self.observation_space.shape)
+        actions = self.policy_tf.step(observation)
+
+        if self.action_noise is not None and not deterministic:
+            actions = np.clip(actions + self.action_noise(), -1, 1)
+
+        actions = actions.reshape((-1,) + self.action_space.shape)  # reshape to the correct action shape
+        actions = actions * np.abs(self.action_space.low)  # scale the output for the prediction
+
+        if not vectorized_env:
+            actions = actions[0]
+
+        return actions, None
+
+    def get_parameter_list(self):
+        return (self.params +
+                self.target_params)
+
+    def save(self, save_path):
+        data = {
+            "learning_rate": self.learning_rate,
+            "buffer_size": self.buffer_size,
+            "learning_starts": self.learning_starts,
+            "train_freq": self.train_freq,
+            "batch_size": self.batch_size,
+            "tau": self.tau,
+            # Should we also store the replay buffer?
+            # this may lead to high memory usage
+            # with all transition inside
+            # "replay_buffer": self.replay_buffer
```

Comment:
I dont think we should, replay buffers are not usually too large, and would quickly be replaced after a few samples.

Not to mention, the `replay_buffer` could hinder transfer learning to a new, but close enough environment.

Worst case, add it to `__init__` as a parameters, with it set to False by default.

Reply:
> and would quickly be replaced after a few samples.

I disagree on that (especially when you have a replay with 1M samples)
but in fact the user can access it anyway (I do that in https://github.com/araffin/learning-to-drive-in-5-minutes/pull/9 for pretraining)

Reply:
> I disagree on that (especially when you have a replay with 1M samples)
Ah yes, I was thinking with images, but indeed with small observations it would be big.